lib/python/amsterdam-schema/amsterdam_schema/types.py

```diff
@@ -72,7 +72,43 @@ def from_dict(cls, obj: dict):
     @property
     def tables(self) -> typing.List[DatasetTableSchema]:
         """Access the tables within the file"""
-        return [DatasetTableSchema(i, _parent_schema=self) for i in self["tables"]]
+        tables = []
+        for table_schema in self["tables"]:
+            table = DatasetTableSchema(table_schema, _parent_schema=self)
+            tables.append(table)
+            for field_name, field in table_schema["schema"]["properties"].items():
+                if field.get("type") == "table":
+                    # Map Arrays into tables.
+                    sub_table_schema = dict(
+                        id=f"{table.id}_{field_name}",
+                        type="table",
+                        schema={
+                            "$schema": "http://json-schema.org/draft-07/schema#",
+                            "type": "object",
+                            "additionalProperties": False,
+                            "required": [
+                                "id",
+                                "schema"
+                            ],
+                            "properties": {
+                                "id": {
+                                    "$ref": "https://schemas.data.amsterdam.nl/schema@v1.1.0#/definitions/id",
```

Comment:
In de schema die we nu gebruiken, maken we het "id" van het type integer of string. De $ref gebruiken we niet meer. De reden is dat $ref het type niet exact vastlegt (kan of een string of een integer worden) en dat staat postgres niet toe.
Wellicht beter om hier ook een concreet type als integer of string te kiezen? (hopelijk is dat mogelijk)